### PR TITLE
PMM-6567 pg_stat_monitor

### PR DESCRIPTION
**Pull request by JiriCtvrtka**

### CODE REVIEW

**Comment by JiriCtvrtka** [comment]
FB: https://github.com/Percona-Lab/pmm-submodules/pull/1104

PRs:
PMM - https://github.com/percona/pmm/pull/584
PMM managed - https://github.com/percona/pmm-managed/pull/457
PMM API tests - https://github.com/Percona-Lab/pmm-api-tests/pull/124
PMM agent - https://github.com/percona/pmm-agent/pull/186

**Comment by AlekSi** [comment]
It should be release/2.11, not release/2.11.0

File: Gopkg.toml [hunk 1]
```diff
@@ -26,7 +26,7 @@ required = [
 
 [[constraint]]
   name = "github.com/percona/pmm"
-  branch = "PMM-2.0"
+  branch = "PMM-6567-pg-monitor"
```
Comment (askomorokhov) [inline Gopkg.toml:29]: 👀

File: agents/postgres/pgstatmonitor/pgstatmonitor.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+			prevPSS = new(pgStatMonitorExtended)
+			count = float32(currentPSS.Calls)
+		case prevPSS.Calls == 0:
+			m.l.Debugf("New query: %s.", currentPSS)
+		default:
+			m.l.Debugf("Normal query: %s.", currentPSS)
+		}
+
+		mb := &agentpb.MetricsBucket{
+			Common: &agentpb.MetricsBucket_Common{
```
Comment (askomorokhov) [inline agents/postgres/pgstatmonitor/pgstatmonitor.go:225]: Have we missed client_host here?

File: agents/postgres/pgstatmonitor/stat_monitor_cache.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+	"time"
+
+	"github.com/percona/pmm-agent/utils/truncate"
+
+	"github.com/AlekSi/pointer"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/go-mysql/query"
```
Comment (askomorokhov) [inline agents/postgres/pgstatmonitor/stat_monitor_cache.go:31]: Shouldn't it be https://github.com/lfittl/pg_query_go here? 
https://github.com/percona/pmm-agent/pull/183#discussion_r488456480

**Review by askomorokhov — CHANGES_REQUESTED** [review]

File: agents/postgres/pgstatmonitor/stat_monitor_cache.go (hunk 3 above)
Comment (JiriCtvrtka) [reply agents/postgres/pgstatmonitor/stat_monitor_cache.go]: Yes, right I clone it from 6460 and there it wasnt still fixed. Thanks.

**Review by askomorokhov — APPROVED** [review]

File: agents/postgres/pgstatmonitor/pgstatmonitor.go [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+			// Currently, we can't differentiate between those situations.
+			m.l.Tracef("Skipped due to the same number of queries: %s.", currentPSM)
+			continue
+		case count < 0:
+			m.l.Debugf("Truncate detected (negative count). Treating as a new query: %s.", currentPSM)
+			prevPSM = new(pgStatMonitorExtended)
+			count = float32(currentPSM.Calls)
+		case now.Sub(currentPSM.BucketStartTime) > queryStatMonitor:
+			// TODO https://jira.percona.com/browse/PMM-6642
+			m.l.Debugf("Truncate detected (old bucket). Treating as a new query: %s.", currentPSM)
```
Comment (AlekSi) [inline agents/postgres/pgstatmonitor/pgstatmonitor.go:220]: @JiriCtvrtka @BupycHuk please keep that separate `case`

**Pull request merged**